### ISSUE TITLE
SEACAS:  Algebra: free(): invalid pointer

### ISSUE THREAD

**Issue by lxmota** [trilinos/Trilinos#7505]
## Bug Report
@trilinos/seacas 

### Description
The algebra executable that is part of seacas fails to execute and aborts with a free(): invalid pointer error.

### Steps to Reproduce
`./algebra outCohesive.e outCohesive.alg.e < alg.i
`

Input files and error log:

[alg.zip](https://github.com/trilinos/Trilinos/files/4755049/alg.zip)
[alg.txt](https://github.com/trilinos/Trilinos/files/4755046/alg.txt)

@ikalash This is causing cohesive element tests failures in LCM.

**Comment by gsjaardema** [comment]
I am taking a look, but it looks like the `alg.i` file has different variable names than are on the file:
```
 Variables Names:
    Global:
    Nodal:    DISP_X                 DISP_Y
              DISP_Z                 FINT_X
              FINT_Y                 FINT_Z
    Element:  COHESIVE_TRACTION_1_1  COHESIVE_TRACTION_2_1
              COHESIVE_TRACTION_3_1  COHESIVE_TRACTION_1_2
              COHESIVE_TRACTION_2_2  COHESIVE_TRACTION_3_2
              COHESIVE_TRACTION_1_3  COHESIVE_TRACTION_2_3
              COHESIVE_TRACTION_3_3  COHESIVE_TRACTION_1_4
              COHESIVE_TRACTION_2_4  COHESIVE_TRACTION_3_4
```
The `alg.i` refers to:
```
traction_x_1 = COHESIVE_TRACTION_01
traction_y_1 = COHESIVE_TRACTION_02
traction_z_1 = COHESIVE_TRACTION_03

traction_x_2 = COHESIVE_TRACTION_04
traction_y_2 = COHESIVE_TRACTION_05
traction_z_2 = COHESIVE_TRACTION_06

traction_x_3 = COHESIVE_TRACTION_07
traction_y_3 = COHESIVE_TRACTION_08
traction_z_3 = COHESIVE_TRACTION_09

traction_x_4 = COHESIVE_TRACTION_10
traction_y_4 = COHESIVE_TRACTION_11
traction_z_4 = COHESIVE_TRACTION_12
```

**Comment by lxmota** [comment]
@gsjaardema Thanks! Yes, these variables changed names months ago, so it would be surprising that only now this shows up. I'll go update the variable names, so it may not be a bug in SEACAS at all.

**Comment by gsjaardema** [comment]
It still shouldn't core dump even with bad variable names.  I did run with current alg.i and one I modified to match variable names using the current SEACAS/github version and it doesn't core dump.  There have been some changes to the underlying Fortran memory managment system since the last Trilinos sync, so maybe that fixed the issue.   

I will try to reproduce on current Trilinos code to see if the issue is fixed, or if it just isn't showing up for me.

**Comment by lxmota** [comment]
Ok, I changed the alg.i file to the below, but I still get the same error.

```
save all

traction_x_1 = COHESIVE_TRACTION_1_1
traction_y_1 = COHESIVE_TRACTION_2_1
traction_z_1 = COHESIVE_TRACTION_3_1

traction_x_2 = COHESIVE_TRACTION_1_2
traction_y_2 = COHESIVE_TRACTION_2_2
traction_z_2 = COHESIVE_TRACTION_3_2

traction_x_3 = COHESIVE_TRACTION_1_3
traction_y_3 = COHESIVE_TRACTION_2_3
traction_z_3 = COHESIVE_TRACTION_3_3

traction_x_4 = COHESIVE_TRACTION_1_4
traction_y_4 = COHESIVE_TRACTION_2_4
traction_z_4 = COHESIVE_TRACTION_3_4

delete COHESIVE_TRACTION_1_1, COHESIVE_TRACTION_2_1, COHESIVE_TRACTION_3_1, COHESIVE_TRACTION_1_2
delete COHESIVE_TRACTION_2_2, COHESIVE_TRACTION_3_2, COHESIVE_TRACTION_1_3, COHESIVE_TRACTION_2_3
delete COHESIVE_TRACTION_3_3, COHESIVE_TRACTION_1_4, COHESIVE_TRACTION_2_4, COHESIVE_TRACTION_3_4

exit
```

**Comment by gsjaardema** [comment]
This should be fixed by #7513
